src/Agent.Sdk/Knob/Knob.cs

```diff
@@ -24,10 +24,17 @@ public ExperimentalKnob(string name, string description, params IKnobSource[] so
         }
     }
 
+    public class SecretKnob : Knob
```

Comment:
I don't think we need a derived class for this. Can we just add `isSecret` property to the existing Knob class? :thinking:  @alexander-smolyakov could you please take a look?

Reply:
We have Deprecated and Experimantal subclass for similar goals. Why can't we use subclass in this case?

Reply:
I think a class property is generally better than inheriting a class.
Yes, we already have `DeprecatedKnob` and `ExperimentalKnob` that work by inheritance, but this will get harder to maintain if we want to add more types of knobs. E.g. an experimental knob can be a secret one at the same time.
So it would be better to add a class property here.

Reply:
```SecretAndExperimental``` subclass would be in this case

Reply:
> I don't think we need a derived class for this. Can we just add isSecret property to the existing Knob class? 🤔 @alexander-smolyakov could you please take a look?

sure, let me take a look.

Reply:
Considering the fact that we are not planning to mark every Knob as the secret, I think the current implementation is good enough.